x-pack/plugins/actions/server/builtin_action_types/case/utils.ts

```diff
@@ -0,0 +1,253 @@
+    throw new Error('Instance is not alive.');
+  }
+};
+
+export const request = async ({
+  axios,
+  url,
+  method = 'get',
+  data = {},
+}: {
+  axios: AxiosInstance;
+  url: string;
+  method?: Method;
+  // This will have to remain `any` until we can extend connectors with generics
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data?: any;
+}): Promise<AxiosResponse> => {
+  const res = await axios(url, { method, data });
+  throwIfNotAlive(res.status, res.headers['content-type']);
+  return res;
+};
+
+export const patch = ({
+  axios,
+  url,
+  data,
+}: {
+  axios: AxiosInstance;
+  url: string;
+  // This will have to remain `any` until we can extend connectors with generics
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data: any;
+}): Promise<AxiosResponse> => {
+  return request({
+    axios,
+    url,
+    method: 'patch',
+    data,
+  });
+};
+
+export const addTimeZoneToDate = (date: string, timezone = 'GMT'): string => {
+  return `${date} ${timezone}`;
+};
+
+export const prepareFieldsForTransformation = ({
+  params,
+  mapping,
+  defaultPipes = ['informationCreated'],
+}: PrepareFieldsForTransformArgs): PipedField[] => {
+  return (
+    Object.keys(params.externalCase)
+      // This clears undefined values but typescript does not get it
+      .filter(p => mapping.get(p)?.actionType)
+      .filter(p => mapping.get(p)?.actionType !== 'nothing')
+      .map(p => ({
+        key: p,
+        value: params.externalCase[p],
+        actionType: mapping.get(p)?.actionType ?? 'nothing',
+        pipes: [...defaultPipes],
+      }))
+      .map(p => ({
+        ...p,
+        pipes: p.actionType === 'append' ? [...p.pipes, 'append'] : p.pipes,
+      }))
+  );
+};
+
+const t = { ...transformers } as { [index: string]: Function }; // TODO: Find a better solution if exists.
+
+export const transformFields = ({ params, fields, currentIncident }: TransformFieldsArgs) => {
+  return fields.reduce((prev, cur) => {
+    const transform = flow(...cur.pipes.map(p => t[p]));
+    prev[cur.key] = transform({
+      value: cur.value,
+      date: params.updatedAt ?? params.createdAt,
+      user:
+        params.updatedBy != null
+          ? params.updatedBy.fullName
+            ? params.updatedBy.fullName
+            : params.updatedBy.username
+          : params.createdBy.fullName
+          ? params.createdBy.fullName
+          : params.createdBy.username,
+      previousValue: currentIncident ? currentIncident[cur.key] : '',
+    }).value;
+    return prev;
+    // This will have to remain `any` until we can extend connectors with generics
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  }, {} as any);
+};
+
+export const transformComments = (comments: Comment[], pipes: string[]): Comment[] => {
+  return comments.map(c => ({
+    ...c,
+    comment: flow(...pipes.map(p => t[p]))({
```

Comment:
if you apply all the refactor above do not forget to do that
`comment: flow(...pipes.map(p => transformers[p]))({`